reducers/chatsReducer.js

```diff
@@ -14,38 +14,54 @@ function chatsReducer(state = chatsInitialState, action) {
                 ...state,
                 chats: payload
             };
+
         case types.SET_CURRENT_CHAT:
             return {
                 ...state,
                 currentChat: payload
             };
+
         case types.CREATE_CHAT:
             return {
                 ...state,
                 chats: state.chats.concat(payload)
             };
+
         case types.CREATE_USER_IN_CHAT:
             const { id, display_name: displayName, cid } = payload;
-            const { chats } = state;
+            
             const user = {
                 id,
                 displayName
             };
-            // Concat user object to the users array of assoc. chat
-            const cidIndex = chats.findIndex(
-                (item) => item.id === cid
+
+            const cidIndex = state.chats.findIndex(
+                (chat) => chat.id === cid
             );
+
+            // make copy of chat to update
+            const chatToUpdate = Object.assign({}, state.chats[cidIndex]);
+            // concat added user and return new array of users
+            const updatedChatUsers = { users: chatToUpdate.users.concat(user) };
+            // make copy of updated chat to replace given chat in the state
+            const updatedChat = { ...chatToUpdate, ...updatedChatUsers };
+
             return {
                 ...state,
                 addedUser: user,
-                chats: {
-                    ...state.chats,
-                    [chats[cidIndex].users]: [
-                        ...state.chats[cidIndex].users,
-                        user
-                    ]
-                }
+                chats: state.chats.map(
+                    (chat, i) => i === cidIndex ? updatedChat : chat
+                )
```

Comment:
heads up, i changed this so it's using .map instead of using slicing. it just looks like a lot cleaner imo.